### ISSUE TITLE
News and discussion

### ISSUE THREAD

**Issue by skaman82** [skaman82/cleanflight-configurator#1]
Managed to rework some basic layout today. Not looking pretty yet, just basic work. Placed dataflash status bar in the header but can’t figure out how to trigger a readout on connect in serial_backend.js. Some help?

![bildschirmfoto 2015-10-22 um 18 11 56](https://cloud.githubusercontent.com/assets/11768024/10671224/c94d7e3a-78e9-11e5-8f70-19cbe2a5a9b3.png)

**Comment by skaman82** [comment]
The new bits and pieces and the actual design work are located here https://github.com/skaman82/CF_GUI

**Comment by skaman82** [comment]
Getting there.... 
![bildschirmfoto 2015-10-23 um 07 07 39](https://cloud.githubusercontent.com/assets/11768024/10685040/00e07cc4-7955-11e5-86ee-35d99f9432f9.png)

**Comment by ProDrone** [comment]
Maybe there is no data in your DataFlash to present?

I checked out your branch, made some Blackbox recording to the internal flash and then get this with your code:

![screenhunter_62](https://cloud.githubusercontent.com/assets/7206988/10690660/86416cb0-7983-11e5-82c8-3be7dd7b5a9d.jpg)

Good to see you started exploring the code!

Btw. It is preferred to make your changes against the development branch instead of master if you want to make a pull request.

**Comment by skaman82** [comment]
@ProDrone  thanks, but I don't meant the Dataflash tab. That is working fine. I placed a Dataflash status bar in the upper right corner in the header. It would be nice to see the state at any time. 

The problem: this small status bar is not showing anything until I actually click on the Dataflash tab. This event should be happening on connect (in serial_backend.js are already some events placed which are triggered on connect/disponect but I don't get it.
<img width="232" alt="bildschirmfoto 2015-10-23 um 12 56 48" src="https://cloud.githubusercontent.com/assets/11768024/10691074/a4cac264-7986-11e5-8882-6305fe89019e.png">
)

**Comment by skaman82** [comment]
@ProDrone I guess that I run into the same problem when I place the profile switch in the header area next ...

**Comment by ProDrone** [comment]
It seems that you are missing code to update the small Dataflash status bar. It works in the dataflash tab because you used the same id's and class names for the small Dataflash status bar, so they both get updated (with the same code) when the Dataflash tab is opened.

As an example you may look at how the sensors are updated in the header.

GUI programming is still not my comfort zone :smile:

**Comment by skaman82** [comment]
New log functionality
![bildschirmfoto 2015-10-24 um 07 26 13](https://cloud.githubusercontent.com/assets/11768024/10709060/a5a7f108-7a20-11e5-9257-8a2141375e29.png)
![bildschirmfoto 2015-10-24 um 07 26 22](https://cloud.githubusercontent.com/assets/11768024/10709061/a8c87eac-7a20-11e5-831a-f566428b508e.png)

**Comment by ProDrone** [comment]
There sure must be a way to solve this. Is something like that not done in the Dataflash tab? Where the amount of free space is made visible. I prefer an always visible log,

It would be great if the tab selector on the left slides to the left and become very small, until you hover over it. Then you keep the whole area of the page for the relevant elements. It is so easy to come up with thoughts on this, that are (probably) hard to implement :smile:

**Comment by skaman82** [comment]
I did manage to fix the bug that prevented scrolling to the the very bottom of the page while log was open. Your idea is great but, as you say, not so easy to do :) 

I think for most users the Log area is wasted space. Its like the very first FTP client apps. The had always a log area at the top. But no popular FTP client has this area enabled by default these days. Simply because for most people this is cryptic information. Minimizing the log helps to get more space also. Considering to make this optional.

**Comment by enly1** [comment]
Great work guys !

The topic of the log area - i'm not a fan of the slow animated open area and the cosmetic scroll indicator - as you say it will probably be a curious new user or someone that needs the log data specifically - in these cases, a quick open access sounds a good idea - or at least optional animations.

Also, just as an FYI I did notice that the close operation isn't qiute right yet - when you open it, you get the log contents, if you scroll up to read something then close it, you don't get rescrolled to the last line - so see the first line instead. which given the ability to see only one line, seems like it should be the last notification.  Thoughts ?

**Comment by skaman82** [comment]
Can't figure out how get the save button bar to stick to the bottom of the #container element. Normally this could be done simply in CSS with: position:fixed; bottom:0px; but in this special case this doesn't work. The idea is to keep the save button always visible and in the same place independent from the length of the content. Would appreciate some help on this. 

<img width="970" alt="bildschirmfoto 2015-10-29 um 17 10 37" src="https://cloud.githubusercontent.com/assets/11768024/10824393/7f5bc27e-7e60-11e5-9d2d-7fc110785263.png">

**Comment by skaman82** [comment]
@VirtualEnder @thenickdude 
Hi, noticed your work on configurator. Maybe you can have a look here and help out with some little things (in the issue list). Need some help with JS. Thanks! @VirtualEnder like your PID rework!

This is how the current state looks like:
![bildschirmfoto 2015-10-29 um 20 46 44](https://cloud.githubusercontent.com/assets/11768024/10830112/37eb9bda-7e7e-11e5-8738-649430e145f4.png)
:

**Comment by VirtualEnder** [comment]
Looks fantastic man!  Great work!  Since winter is setting in and we'll have more code time I might take a whack at fine tuning some of the CSS and JS stuff for you.  LOVE the design though! Huge improvement.

Also agree with the speed of the animations.  I typically try to make my animations in the 200-300ms range, just from a usability standpoint.  Should be pretty easy to change the speed down. 

You definitely took it to the next level man, my PID rework was an hour hack to solve a problem a user on RCGroups wanted. :D Glad to see you taking the whole thing further.

**Comment by skaman82** [comment]
@VirtualEnder Thanks! Looking forward to your expertise and help.

**Comment by wanderzell** [comment]
Hi, I like what you have done so far. Good to see you have already had progress in implementing the new styles. I can help you on the CSS / JS. Just my suggestion would be to start easy. Move in babysteps and implement new looks on top of existing functionality, avoid adding new, yet in the beginning. That way we could also get the skin changes merged into main cleanflight-configurator more quickly and people working there on the new functionalities would also get hands on the new styles.

**Comment by skaman82** [comment]
@wanderzell Thank you for your support. I personally have issues with things that are half done but I think you might be right with going this in small steps. Merging this at once might be simply to big. So far I didn't touched JS much but changed the CSS and HTML files on a large scale to get this done. The basic UI is almost ready except for two things marked red on my issue list. Getting them done is necessary for the first step IMO. 

Then we could move tab by tab reworking things and when this is finally done we could add new functionality like tooltips, profile switch and the data flash element in the header. I think we could figure out something like a road map for this. What do you think?

**Comment by skaman82** [comment]
@ProDrone Quick question: Dominic asked me to rebase my pull request against the current developer branch. How can I do that? Can not find any settings on that on the website. Should I create a new pull request? Working with github for the first time. Here the pull request: https://github.com/cleanflight/cleanflight-configurator/pull/263

**Comment by skaman82** [comment]
@wanderzell Could you please have a quick look at this element in the config tab if you have time? Somehow i can't klick the switch directly. It only reacts when I click between the switch and the text. All other toggle switches are working as expected. Could not solve this yet.

**Comment by skaman82** [comment]
@wanderzell Forgot to add the image of the area.
<img width="573" alt="bildschirmfoto 2015-11-01 um 15 54 43" src="https://cloud.githubusercontent.com/assets/11768024/10869479/862222bc-80b1-11e5-8060-35b0e3192048.png">

**Comment by VirtualEnder** [comment]
@skaman82 , check the z-index.  Looks like something is overlapping the button.

**Comment by skaman82** [comment]
@VirtualEnder Yes I think you are right but can't figure out what is overlapping. 

Von meinem iPhone gesendet

**Comment by VirtualEnder** [comment]
Right click and click "Inspect element" It'll show you which one is overlapping.

**Comment by skaman82** [comment]
I’ve done that. It shows me the same switchery <span> element as it does on other toggle switches.
Can not see anything strange.

**Comment by wanderzell** [comment]
I can take a look on this later today.

**Comment by VirtualEnder** [comment]
Ok, here's the fixed code:

```
<div class="disarm">
          <div class="checkbox">
            <div style="float:left; height:20px; margin-right:5px;">
                <input type="checkbox" name="disarmkillswitch" id="toggle" data-switchery="true" style="display: none;" class="feature" title="disarmkillswitch"><span class="switchery switchery-default" style="border-color: rgb(89, 170, 41); box-shadow: rgb(89, 170, 41) 0px 0px 0px 8px inset; transition: border 0.4s, box-shadow 0.4s, background-color 1.2s; background-color: rgb(89, 170, 41);"><small style="left: 31px; transition: background-color 0.4s, left 0.2s; background-color: rgb(255, 255, 255);"></small></span>
              </div><label for="disarmkillswitch">
              
              <div class="freelabel i18n-replaced" i18n="configurationDisarmKillSwitch">Disarm motors regardless of throttle value (When arming via AUX channel)</div>
            </label>
          </div>
          <div class="number disarmdelay" style="margin-bottom: 5px; display: none;">
            <label>
              <div class="numberspacer">
                <input type="number" name="autodisarmdelay" min="0" max="60">
              </div>
              <span i18n="configurationAutoDisarmDelay" class="i18n-replaced">Disarm motors after set delay(Seconds) (Requires MOTOR_STOP feature)</span>
            </label>
          </div>
        </div>
```

Also in CSS, set the z-index of `.switchery` to be all the way front.

**Comment by skaman82** [comment]
@VirtualEnder Thank you for looking in to that! Are you sure this fixes the right problem? This apperas to be a different switch. I am talking about this one: 
<img width="435" alt="bildschirmfoto 2015-11-01 um 19 22 47" src="https://cloud.githubusercontent.com/assets/11768024/10870571/242daea6-80ce-11e5-992d-9ce16f9f4e6b.png">

**Comment by VirtualEnder** [comment]
Hah, you're right! But that switch had problems too :D

```
<div class="checkbox">
          <div class="numberspacer">
              <input type="checkbox" name="multiwiicurrentoutput" id="toggle" data-switchery="true" style="display: none;"><span class="switchery switchery-default" style="box-shadow: rgb(196, 196, 196) 0px 0px 0px 0px inset; border-color: rgb(196, 196, 196); transition: border 0.4s, box-shadow 0.4s; background-color: rgb(196, 196, 196);"><small style="left: 0px; transition: background-color 0.4s, left 0.2s;"></small></span>
            </div><label for="multiwiicurrentoutput">
            
            <span i18n="configurationBatteryMultiwiiCurrent" class="i18n-replaced">Enable support for legacy Multiwii MSP current output</span>
          </label>
        </div>
```

THere's the fix for that one.

**Comment by skaman82** [comment]
@VirtualEnder Great! Thank you!

**Comment by skaman82** [comment]
Some more Work done:

![bildschirmfoto 2015-11-05 um 16 49 57](https://cloud.githubusercontent.com/assets/11768024/10973356/63c42326-83dd-11e5-8616-7052a7981d9b.png)

![bildschirmfoto 2015-11-05 um 16 49 52](https://cloud.githubusercontent.com/assets/11768024/10973361/6cae9980-83dd-11e5-822a-fa78738258c4.png)

![bildschirmfoto 2015-11-05 um 16 49 27](https://cloud.githubusercontent.com/assets/11768024/10973371/712906f8-83dd-11e5-9c67-98507153fd27.png)

![bildschirmfoto 2015-11-05 um 16 49 24](https://cloud.githubusercontent.com/assets/11768024/10973382/7a8dc472-83dd-11e5-9d66-b91cd4231324.png)

<img width="477" alt="bildschirmfoto 2015-11-05 um 16 37 53" src="https://cloud.githubusercontent.com/assets/11768024/10973385/829f9352-83dd-11e5-9e53-acdaa6303b02.png">

**Comment by skaman82** [comment]
@VirtualEnder I just noticed that the name for the Sonar "PID" is missing in your version of the PID Tab. Is that supposed to be like that?

<img width="509" alt="bildschirmfoto 2015-11-05 um 20 56 49" src="https://cloud.githubusercontent.com/assets/11768024/10979993/082cadc6-8400-11e5-802c-d544a4826eff.png">

**Comment by skaman82** [comment]
New landing page. Change log is now hidden.
![bildschirmfoto 2015-11-07 um 06 21 04](https://cloud.githubusercontent.com/assets/11768024/11013690/de1b7062-8517-11e5-8732-18ac02d98dd4.png)
![Uploading Bildschirmfoto 2015-11-07 um 06.21.07.png…]()

**Comment by skaman82** [comment]
@digitalentity I'm finished with the first phase of configurator GUI rework. Check this branch and see for yourself. I hope you like it. PR is ready to merge. Now I have more time again to test your NAV rewrite :)

**Comment by digitalentity** [comment]
Cool, cool, cooooool! I can't say how cool this is. Finally we have a nicer looping and user-friendlier configurator!

**Comment by marksev1** [comment]
Great job on the UI design! Will it still be a Chrome app? Hopefully once this Chrome will be replaced since the Google guys constantly introduce bugs :D

**Comment by skaman82** [comment]
@marksev1 yes, this is still a chrome app. I just reworked the old app visually.

**Comment by wanderzell** [comment]
Created PR #30 with the reworked styles for landing page. The work was primarily initiated to solve a rendering glitch but ended up being complete rewrite of styles. I also changed a look a bit - made it bit lighter, added bit air into change log. But the overall layout remained the same.

<img width="1106" alt="screen shot 2015-11-10 at 10 37 26" src="https://cloud.githubusercontent.com/assets/4578152/11058127/39761b5a-8799-11e5-9bc2-3b5302bb83de.png">

<img width="1105" alt="screen shot 2015-11-10 at 10 37 59" src="https://cloud.githubusercontent.com/assets/4578152/11058130/3ea6c462-8799-11e5-916d-ee93b3a62e49.png">

Personally I think the change log should NOT be hidden, as it is useful piece of information when you can see it without any prior action. However, I was not able to come up something better so I left it as it is - I think we should come back to it some moment later.

**Comment by skaman82** [comment]
Started to work on new stuff (new_stuff branch) and quickly run into some problems with my JS skills. Placed the dataflash  bar and profile switch in the header and done some groundwork. 

Here the problems:
• Somehow the dataflash and profile status getting updated only on the SECOND connection attempt.
• Dataflash should show the div .noflash and hide div .dataflash-contents when dataflash is not supported
• Profile switch changes the profile but it seems to do it several times
• After changing the profile the current tab should be reinitialised

Placed the JS code in serial_backend.js for now, but I'm sure it has to go in main.js or in gui.js

@VirtualEnder If you have some time, could you help me to sort this out?

**Comment by marksev1** [comment]
Skaman whats the status regarding 1024*600 support?

**Comment by skaman82** [comment]
@marksev1 check the branch GUI_optimization2

**Comment by skaman82** [comment]
@marksev1 just modified some stuff, should work even better now. Feedback is welcome!

**Comment by marksev1** [comment]
Cool I'll try and let you know! :)

**Comment by marksev1** [comment]
Skaman it still seems a bit too big for my micro-netbook, see here: http://imgur.com/a/cBiFg

Probably because panel is taking some space. I'm on linux btw.

**Comment by skaman82** [comment]
@marksev1 you actually got 550px of screen space (600 - panel - app-panel at the top. Just changed the min-with to 550px im my branch.

**Comment by wanderzell** [comment]
@marksev1 @skaman82 I guess part of the problem is setting somewhere that does not allow the app window size be smaller than certain amount of pixels. So your OS shows the bottom menubar over the app window hence making the toolbar unusable. I would remove this minimum window height constraint, then make the left menu show only relevant "tabs" (eg. ones for enabled features only) and it will work. Theoretically at least.

**Comment by skaman82** [comment]
@wanderzell this is a very good idea. I thought in the future the app can be resized even to run on a smartphone (somebody got this to work already btw). Then only a handful of important tabs will be displayed.

Could you please look at my comment regarding some bugs from tho hours ago? Maybe you also have an idea how to solve them :)

**Comment by marksev1** [comment]
Great now I can push the Save button, thanks, http://imgur.com/pNn3LrV

**Comment by VirtualEnder** [comment]
@skaman82 I'll take a look at the JS if I can today.

Have you considered using media queries to scale things down for smaller screens?  You should be able to significantly re-arrange stuff to make it fit without compromising the design of the larger screens.

**Comment by skaman82** [comment]
@VirtualEnder I am already using media queries for some things (eg. for minimizing the navigation if the window is small) This could be done more extensively in the future.
